pkg/cli/admin/rebootmachineconfigpool/reboot_machine_config_pool.go

```diff
@@ -0,0 +1,200 @@
+	machineConfigResource = schema.GroupVersionResource{Group: "machineconfiguration.openshift.io", Version: "v1", Resource: "machineconfigs"}
+
+	//go:embed restart-template.json
+	restartTemplateJSON []byte
+	restartTemplate     = readMachineConfigOrDie(restartTemplateJSON)
+)
+
+func readMachineConfigOrDie(objBytes []byte) *MachineConfig {
+	unstructuredObj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, objBytes)
+	if err != nil {
+		panic(err)
+	}
+
+	uncastObj, ok := unstructuredObj.(*unstructured.Unstructured)
+	if !ok {
+		panic(fmt.Sprintf("unexpected type %T", unstructuredObj))
+	}
+	machineConfig := &MachineConfig{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(uncastObj.Object, machineConfig); err != nil {
+		panic(err)
+	}
+
+	return machineConfig
+}
+
+const (
+	RebootMachineConfigPoolFieldManager = "reboot-machine-config-pool"
+)
+
+type RebootMachineConfigPoolRuntime struct {
+	ResourceFinder genericclioptions.ResourceFinder
+	DynamicClient  dynamic.Interface
+
+	dryRun bool
+
+	Printer printers.ResourcePrinter
+
+	genericclioptions.IOStreams
+}
+
+func (r *RebootMachineConfigPoolRuntime) Run(ctx context.Context) error {
+	visitor := r.ResourceFinder.Do()
+
+	// TODO need to wire context through the visitorFns
+	err := visitor.Visit(r.rebootFromResourceInfo)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *RebootMachineConfigPoolRuntime) rebootFromResourceInfo(info *resource.Info, err error) error {
+	if err != nil {
+		return err
+	}
+
+	if machineConfigPoolKind != info.Object.GetObjectKind().GroupVersionKind() {
+		return fmt.Errorf("command must only be pointed at machineconfigpools")
+	}
+
+	uncastObj, ok := info.Object.(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("not unstructured: %w", err)
+	}
+	machineConfigPool := &MachineConfigPool{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(uncastObj.Object, machineConfigPool); err != nil {
+		return fmt.Errorf("not a secret: %w", err)
+	}
+
+	return r.rebootMachineConfigPool(context.TODO(), machineConfigPool)
+}
+
+func (r *RebootMachineConfigPoolRuntime) createOptions() metav1.CreateOptions {
+	if r.dryRun {
+		return metav1.CreateOptions{
+			DryRun:       []string{metav1.DryRunAll},
+			FieldManager: RebootMachineConfigPoolFieldManager,
+		}
+	}
+	return metav1.CreateOptions{
+		FieldManager: RebootMachineConfigPoolFieldManager,
+	}
+}
+
+func (r *RebootMachineConfigPoolRuntime) updateOptions() metav1.UpdateOptions {
+	if r.dryRun {
+		return metav1.UpdateOptions{
+			DryRun:       []string{metav1.DryRunAll},
+			FieldManager: RebootMachineConfigPoolFieldManager,
+		}
+	}
+	return metav1.UpdateOptions{
+		FieldManager: RebootMachineConfigPoolFieldManager,
+	}
+}
+
+func (r *RebootMachineConfigPoolRuntime) rebootMachineConfigPool(ctx context.Context, machineConfigPool *MachineConfigPool) error {
+	// do something better.
+	if machineConfigPool.Spec.MachineConfigSelector == nil {
```

Comment:
check these

Reply:
So there's two things to this:

1. the `master` and `worker` pools will always be required be in a cluster (even something like SNO, it would simply have an empty worker pool), strictly named (and labelled) as such
2. all custom pools must inherit from `worker` pool, so by updating worker pool, you also roll out any custom pool e.g. `infra` you may have

So to simply this for us, I wonder if it may be easier to only allow specifying `worker` and `master` and strictly match it as such here? A user rebooting `infra` pool directly isn't very valuable, since if they reboot `worker` they will just reboot it again.